src/gsts3multipartuploader.cpp

```diff
@@ -160,6 +160,35 @@ static bool is_null_or_empty(const char* str)
     return str == nullptr || strcmp(str, "") == 0;
 }
 
+static const Aws::String get_bucket_from_config(const GstS3UploaderConfig * config)
+{
+    if (is_null_or_empty(config->location)) {
+        return config->bucket;
+    } else {
+        GstUri *uri = gst_uri_from_string(config->location);
+        Aws::String bucket(gst_uri_get_host(uri));
+        gst_uri_unref(uri);
+        return bucket;
+    }
+}
+
+static const Aws::String get_key_from_config(const GstS3UploaderConfig * config)
+{
+    if (is_null_or_empty(config->key)) {
```

Comment:
I think this is meant to be if `is_null_or_empty(config->location)` as is done in `get_bucket_from_config`